lib/common/common/NumberUtilities.cpp

```diff
@@ -76,8 +76,14 @@ namespace Js
         return (ulong)llu;
 #endif //!I386_ASM
 
-#else
-#error Neither _WIN32, nor _WIN64 is defined
+#elif defined(_M_X64) //!WINDOWS, 64bit
+        __uint128_t result = (__uint128_t)lu1 * (__uint128_t)lu2;
```

Comment:
`pal/inc/rt/intsafe.h` implements `UInt32x32To64`, is there any reason we don't use the one defined there?